tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/ScalaJSClassEmitter.scala

```diff
@@ -232,7 +273,15 @@ private[scalajs] final class ScalaJSClassEmitter(
             List(js.This()))
       }
       val fieldDefs = genFieldDefs(tree)
-      js.Function(Nil, js.Block(superCtorCall :: fieldDefs))
+      if (!usesJSConstructorOpt(className)) {
+        js.Function(Nil, js.Block(superCtorCall :: fieldDefs))
+      } else {
+        // only handling the case for one constructor so far
+        val (ctor :: Nil) =
```

Comment:
There's another occurence that needs that change in that file, I'm changing it too.